src/compiler/checker.ts

```diff
@@ -33694,8 +33694,18 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         }
 
         function getInstantiatedSignatures(signatures: readonly Signature[]) {
-            const applicableSignatures = filter(signatures, sig => !!sig.typeParameters && hasCorrectTypeArgumentArity(sig, typeArguments));
-            return sameMap(applicableSignatures, sig => {
+            const sameTypeAritySignatures = filter(signatures, sig => !!sig.typeParameters && hasCorrectTypeArgumentArity(sig, typeArguments));
+            if (!sameTypeAritySignatures.length) {
+                return sameTypeAritySignatures;
+            }
+            const applicableSignatures = mapDefined(sameTypeAritySignatures, sig => {
+                const typeArgumentTypes = checkTypeArguments(sig, typeArguments!, /*reportErrors*/ false);
+                return typeArgumentTypes && getSignatureInstantiation(sig, typeArgumentTypes, isInJSFile(sig.declaration));
+            });
+            if (applicableSignatures.length) {
+                return applicableSignatures;
+            }
+            return sameMap(sameTypeAritySignatures, sig => {

```

Comment:
I've preserved the old behavior here~. If there are some signatures for the given type argument arity but when none of them match... I still gonna report the same error. Perpahs this can be improved further.